lib/Parse/ParseStmt.cpp

```diff
@@ -62,6 +62,7 @@ bool Parser::isStartOfStmt() {
   case tok::pound_warning:
   case tok::pound_error:
   case tok::pound_sourceLocation:
+    if (peekToken().is(tok::colon)) return false;
```

Comment:
`default:` is a valid statement.

Reply:
Reverted in https://github.com/apple/swift/pull/22714/commits/faecb8d5f15a3421f661bd606158d0394dbb19be because the logic is now in `parseList` as you described.